solutions/remotemonitoring/single-vm/setup.sh

```diff
@@ -72,6 +72,26 @@ export APPLICATION_SECRET=$PCS_APPLICATION_SECRET
 
 # ========================================================================
 
+### Install Docker
+
+install_docker_ce() {
+    apt-get update
+    # Remove old packages if installed
+    set +e
+    apt-get remove docker docker-engine docker.io
+    set -e
+    # Install Docker's GPG key
+    apt-get -y --force-yes --no-install-recommends install apt-transport-https ca-certificates curl gnupg2 software-properties-common
+    curl -fsSL https://download.docker.com/linux/$(. /etc/os-release; echo "$ID")/gpg | sudo apt-key add -
+    # Add Docker repository to get up to date packages
+    add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/$(. /etc/os-release; echo "$ID") $(lsb_release -cs) stable"
+    apt-get update
+    # Install
+    apt-get -y install docker-ce docker-compose
+    # Test
+    docker run --rm hello-world && docker rmi hello-world
```

Comment:
why do you need to pull & run hello-world?  i.e. what happen if it fails?  Will the setup.sh script be run again?

Reply:
My guess: since copied from the DS folder, is that this method will exit with error code that gets bubbled up and fails the deployment. There is no re-try (and we didn't have it DockerExtension either). But would be a nice improvement to add at a later time